### PR TITLE
doc: Added links to RST files in header files

### PR DESCRIPTION
**Pull request by richabp**
Provided comments in the header files about the location of corresponding RST files
 and rendered documentation for the particular library.

Ref: NCSDK-11624

Signed-off-by: Richa Pandey <richa.pandey@nordicsemi.no>

### CODE REVIEW

File: include/app_event_manager_profiler_tracer.h [hunk 1]
```diff
@@ -4,6 +4,13 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+/*
+ * The RST file for this library can be found in
+ * doc/nrf/libraries/others/event_manager_profiler_tracer.rst.
+ * Rendered documentation is available at
+ * https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/others/event_manager_profiler_tracer.html.
```
Comment (b-gent) [inline include/app_event_manager_profiler_tracer.h:11]:
```suggestion
 * https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/others/app_event_manager_profiler_tracer.html.
```

File: include/app_event_manager_profiler_tracer.h [hunk 2]
```diff
@@ -4,6 +4,13 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+/*
+ * The RST file for this library can be found in
+ * doc/nrf/libraries/others/event_manager_profiler_tracer.rst.
```
Comment (b-gent) [inline include/app_event_manager_profiler_tracer.h:9]:
```suggestion
 * doc/nrf/libraries/others/app_event_manager_profiler_tracer.rst.
```

**Comment by b-gent** [comment]
Leaving a few comments. Also the comments are missing from the following headers:
app_event_manager.h
event_manager_proxy.h
qos.h

**Review by b-gent — CHANGES_REQUESTED** [review]

File: include/app_event_manager_profiler_tracer.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (jenseneducation-richa-bajpai) [reply include/app_event_manager_profiler_tracer.h]: It has 80 characters limit per line so had to break it where it was exceeding.

File: include/gzp_config.h [hunk 3]
```diff
@@ -4,6 +4,11 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+/*
+ * The RST file for this library can be found in doc/nrf/libraries/gazell/gzp.rst.
```
[the remark this answers is not in the document]
Comment (jenseneducation-richa-bajpai) [reply include/gzp_config.h]: Done.

File: include/app_event_manager_profiler_tracer.h (hunk 1 above)
Comment (jenseneducation-richa-bajpai) [reply include/app_event_manager_profiler_tracer.h]: Done.

File: include/app_event_manager_profiler_tracer.h (hunk 2 above)
Comment (jenseneducation-richa-bajpai) [reply include/app_event_manager_profiler_tracer.h]: Done.

[the remark this answers is not in the document]
Comment (richabp) [reply include/app_event_manager_profiler_tracer.h]: It has 80 characters limit per line so had to break it.

File: include/gzp_config.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (richabp) [reply include/gzp_config.h]: Done.

File: include/app_event_manager_profiler_tracer.h (hunk 1 above)
Comment (richabp) [reply include/app_event_manager_profiler_tracer.h]: Done

File: include/app_event_manager_profiler_tracer.h (hunk 2 above)
Comment (richabp) [reply include/app_event_manager_profiler_tracer.h]: Done

**Comment by gmarull** [comment]
> I personally disagree with this.

Agreed, this is going to be a maintenance hell

**Pull request closed without merging**